RIOT-OS/RIOT
# dist/tools/mspdebug: build from source

### Contribution description

This adds mspdebug as package, similar to EDBG, so that the programmer/debugger is build from source.

This has the advantage that we can indeed provide patches of our own. The first patch fixes a bug with the CPU detection of `mspdebug` in combination with the Olimex MSP430-JTAG-TINY-V2. The second adds the `--expect-id <CPU_NAME>` argument.

The RIOT integration is updated to directly make use of the `--expect-id` parameter. No more spending time debugging why firmware the firmware for the `olimex-msp430-h2618` doesn't run when flashed on the `olimex-msp430h1611` hardware :D

### Testing procedure

Connect an MSP430 board to the Olimex MSP430-JTAG-TINY-V2, subsequently named `CORRECT_MSP430_BOARD`. **DO NOT** connect `WRONG_MSP430_BOARD` (any other MSP430 board but the one connected).

The following should now fail with this PR, while it won't fail on `master`:

```
make BOARD=<WRONG_MSP430_BOARD> -C examples/default flash
make BOARD=<WRONG_MSP430_BOARD> -C examples/default debug-server
make BOARD=<WRONG_MSP430_BOARD> -C examples/default debug
```

The following should still work (did also work on `master`):

```
make BOARD=<CORRECT_MSP430_BOARD> -C examples/default flash
make BOARD=<CORRECT_MSP430_BOARD> -C examples/default debug-server
make BOARD=<CORRECT_MSP430_BOARD> -C examples/default debug
```

Finally, flashing and debugging should work even without `mspdebug` installed locally.

### Issues/PRs references

None
